### PR TITLE
Add __repr__ to TreeSequence

### LINKED ISSUE #938: make print(ts) give an informative message
A quick note, as someone asked this during a workshop. Maybe something like
```
print(f"This is a tree sequence on a sequence of {ts.sequence_length} with {ts.num_samples} samples. It has {ts.num_trees} trees and {ts.num_sites} segregating sites.")
```
... but less verbose.

### PR DESCRIPTION
**Pull request by benjeffery**
## Description

Displays a table-based summary of the tree sequence.

Screenshot from 2020-11-11 10-06-36

Fixes #938 

# PR Checklist:

- [X] Tests that fully cover new/changed functionality.
- [ ] Documentation including tutorial content if appropriate.
- [x] Changelogs, if there are API changes.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/985/)

File: python/tskit/util.py [hunk 1]
```diff
@@ -316,6 +316,38 @@ def obj_to_collapsed_html(d, name=None, open_depth=0):
         return f"{name} {d}"
 
 
+def unicode_table(title, header, rows):
+    if header:
```
Comment (jeromekelleher) [inline python/tskit/util.py:320]: ``if header is not None`` I'd say?

File: python/tskit/util.py [hunk 2]
```diff
@@ -316,6 +316,38 @@ def obj_to_collapsed_html(d, name=None, open_depth=0):
         return f"{name} {d}"
 
 
+def unicode_table(title, header, rows):
```
Comment (jeromekelleher) [inline python/tskit/util.py:319]: A quick docstring would help here. (no need for full sphinx param descriptions, etc)

File: python/tskit/util.py [hunk 3]
```diff
@@ -316,6 +316,38 @@ def obj_to_collapsed_html(d, name=None, open_depth=0):
...
+    else:
+        all_rows = rows
+    widths = [
+        max(len(row[i_col]) for row in all_rows) for i_col in range(len(all_rows[0]))
+    ]
+    out = []
+    if title is not None:
+        w = sum(widths) + len(rows[1]) - 1
+        out += [
+            f"╔{'═' * w}╗\r\n" f"║{title.ljust(w)}║\r\n",
```
Comment (jeromekelleher) [inline python/tskit/util.py:331]: Why the ``\r``s here? I think Python handles the universal newlines stuff and we can just put in ``\n``.

File: python/tskit/trees.py [hunk 4]
```diff
@@ -3334,6 +3334,33 @@ def dump_text(
...
+                        table.num_rows,
+                        "TODO",
+                        "Yes"
+                        if hasattr(table, "metadata") and len(table.metadata) > 0
+                        else "No",
+                    ]
+                ]
+            )
+        return util.unicode_table(
+            "TSKIT TreeSequence", None, ts_rows
```
Comment (jeromekelleher) [inline python/tskit/trees.py:3361]: I'm not sure about the "TSKIT" all-caps here, feels a bit Trumpian. I think we can just say "TreeSequence", right?

File: python/tskit/trees.py [hunk 5]
```diff
@@ -3334,6 +3334,33 @@ def dump_text(
                 )
                 print(row, file=provenances)
 
+    def __repr__(self):
+        ts_rows = [
+            ["Trees", str(self.num_trees)],
+            ["Seq Length", str(self.sequence_length)],
```
Comment (jeromekelleher) [inline python/tskit/trees.py:3340]: We can probably say "Sequence Length", I think there's plenty hspace.

**Review by jeromekelleher — APPROVED** [review]
Love it! This is super cool.

File: python/tskit/util.py (hunk 1 above)
Comment (benjeffery) [reply python/tskit/util.py]: Fixed

File: python/tskit/util.py (hunk 2 above)
Comment (benjeffery) [reply python/tskit/util.py]: Done. Also added a docstring for the HTML method above this one.

File: python/tskit/util.py (hunk 3 above)
Comment (benjeffery) [reply python/tskit/util.py]: Cool! Removed them all.

File: python/tskit/trees.py (hunk 4 above)
Comment (benjeffery) [reply python/tskit/trees.py]: Sorry been reading too much twitter. Fixed.

File: python/tskit/trees.py (hunk 5 above)
Comment (benjeffery) [reply python/tskit/trees.py]: Done

**Comment by benjeffery** [comment]
@jeromekelleher fixed up.

**Comment by petrelharp** [comment]
Wow!!!!!!!!

**Pull request merged**